File: front-end/src/api/domain.ts
/** 도메인 가용성 확인 API */

import { api } from './client'
import type { DomainCheckRequest, DomainCheckResult } from './types'

const PREFIX = '/v1/domain'

/** 단건 도메인 가용성 확인 */
export function checkDomain(body: DomainCheckRequest) {
  return api.post<DomainCheckResult>(`${PREFIX}/check`, body)
}

File: back-end/app/api/domain.py
"""도메인 가용성 확인 API 엔드포인트"""

from fastapi import APIRouter, Depends

from app.plugins.nhn_domain_plugin import NhnDomainPlugin
from app.schemas.domain import DomainCheckRequest, DomainCheckResult
from app.services.domain_service import DomainService

router = APIRouter(prefix="/domain", tags=["domain"])


def get_domain_service() -> DomainService:
    """DomainService DI"""
    from app.main import app_state

    return DomainService(nhn_domain=app_state["nhn_domain"])


@router.post("/check", summary="도메인 가용성 확인", response_model=DomainCheckResult)
async def check_domain(
    request: DomainCheckRequest,
    service: DomainService = Depends(get_domain_service),
):
    """NHN Cloud API를 통한 도메인 등록 가능 여부 확인"""
    return await service.check(request.domain_name)
